server/data/taskClient.ts

```diff
@@ -27,20 +27,24 @@ export default class TaskClient {
     page,
     sortDirection,
     sortBy,
-    taskType,
+    taskTypes,
   }: {
     allocatedFilter: AllocatedFilter
     apAreaId: ApArea['id']
     allocatedToUserId: string
     page: number
     sortDirection: string
     sortBy: TaskSortField
-    taskType?: TaskType
+    taskTypes?: Array<TaskType>
   }): Promise<PaginatedResponse<Task>> {
+    const filters = {} as Record<string, string>
+    if (taskTypes.length) {
+      filters.types = taskTypes.join(',')
+    }
     return this.restClient.getPaginatedResponse({
       path: paths.tasks.index.pattern,
       page: page.toString(),
-      query: { allocatedFilter, allocatedToUserId, apAreaId, sortBy, sortDirection, taskType },
+      query: { ...filters, allocatedFilter, allocatedToUserId, apAreaId, sortBy, sortDirection },
     })
   }
```

Comment:
I think we suggested moving the query out and then spreading it below when we discussed [this suggestion](https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/1531#discussion_r1504217526) earlier, but I've just had a quick look at this locally and I think I see the typing issue you were coming up against. The code below is an alternative that avoids having a `filters` object containing only one of the multiple filters

```ts
    const query = {
      allocatedFilter,
      allocatedToUserId,
      apAreaId,
      sortBy,
      sortDirection,
      types: taskTypes.join(',') || undefined,
    }

    return this.restClient.getPaginatedResponse({
      path: paths.tasks.index.pattern,
      page: page.toString(),
      query,
    })
```

There are few other ways this could be done without introducing the partial `filters` object, but what do you think about this approach?

In any case, this is less important than the commit message, so if you're able to sort the commit message out I'm happy to leave this one as is for now

Reply:
with this approach we are sending undefined for types to the api and we need to remove again that

Reply:
Yeah I wondered is that was an issue 😅

This is probably the simplest change to improve the clarity of labels while avoiding sending `undefined` over then? Essentially moving all the filters into the `filters` object

```ts
    const filters = { allocatedFilter, allocatedToUserId, apAreaId } as Record<string, string>
    if (taskTypes.length) {
      filters.types = taskTypes.join(',')
    }
    return this.restClient.getPaginatedResponse({
      path: paths.tasks.index.pattern,
      page: page.toString(),
      query: { ...filters, sortBy, sortDirection },
    })
```

Reply:
the existing one is a bit cleaner i suppose

Reply:
Though I think it would be better for all filters to go in the `filters` object rather than just `filtersThatNeedFurtherManipulation`/`justTypeFilters`, this is a minor thing so I'm happy to leave it for now